src/Rules.Framework/RulesEngine.cs

```diff
@@ -68,6 +70,24 @@ public Task<RuleOperationResult> AddRuleAsync(Rule<TContentType, TConditionType>
             return this.AddRuleInternalAsync(rule, ruleAddPriorityOption);
         }
 
+        /// <summary>
+        /// Deactivates the specified existing rule.
+        /// </summary>
+        /// <param name="rule">The rule.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException">rule</exception>
+        public Task<RuleOperationResult> DeactivateRuleAsync(Rule<TContentType, TConditionType> rule)
+        {
+            if (rule is null)
+            {
+                throw new ArgumentNullException(nameof(rule));
+            }
+
```

Comment:
don't we want to validate the rule here, as we do in UpdateRuleAsync ?

Reply:
I agree, added the validation